homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,179 @@
+"""Home Assistant Switcher Component Switch platform."""
+
+from logging import getLogger
+from typing import Any, Callable, cast, Dict, Optional
+
+from homeassistant.components.switch import ATTR_CURRENT_POWER_W, SwitchDevice
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.exceptions import PlatformNotReady
+
+from . import (
+    ATTR_AUTO_OFF_SET, ATTR_DEVICE_NAME, ATTR_ELECTRIC_CURRNET,
+    ATTR_REMAINING_TIME, DATA_DEVICE, DOMAIN, SIGNAL_SWITCHER_DEVICE_UPDATE)
+
+_LOGGER = getLogger(__name__)
+
+DEPENDENCIES = ['switcher_kis']
+
+PROPERTIES_TO_DEVICE_ATTRIBUTES = {
+    'current_power_w': ATTR_CURRENT_POWER_W,
+    'electric_current': ATTR_ELECTRIC_CURRNET,
+    'remaining_time': ATTR_REMAINING_TIME,
+    'auto_off_set': ATTR_AUTO_OFF_SET,
+    'device_name': ATTR_DEVICE_NAME
+}
+
+
+async def async_setup_platform(hass: HomeAssistantType, config: Dict,
+                               async_add_entities: Callable,
+                               discovery_info: Dict) -> None:
+    """Set up the switcher platform for the switch component."""
+    if DOMAIN not in hass.data:
+        raise PlatformNotReady("No configuration data found.")
+
+    async_add_entities([SwitcherControl(hass.data[DOMAIN][DATA_DEVICE])])
+
+
+class SwitcherControl(SwitchDevice):
+    """Home Assistant switch entity."""
+
+    def __init__(self, device_data: Any) -> None:
+        """Initialize the entity."""
+        # pylint: disable=unused-import
+        from aioswitcher.devices import SwitcherV2Device
+        # pylint: enable=unused-import
+        self._self_initiated = False
+        self._device_data = device_data  # type: SwitcherV2Device
+        self._state = device_data.state
+        self._unsub_dispatcher = None  # type: Optional[Callable[..., Any]]
+
+    @property
+    def electric_current(self) -> float:
+        """Return the electric current."""
+        return cast(float, self._device_data.electric_current)
+
+    @property
+    def remaining_time(self) -> str:
+        """Return the remaining time to off command."""
+        return cast(str, self._device_data.remaining_time)
+
+    @property
+    def auto_off_set(self) -> str:
+        """Return the auto off configuration set."""
+        return cast(str, self._device_data.auto_off_set)
+
+    @property
+    def device_name(self) -> str:
```

Comment:
Refactor to property `name` and remove it from state attributes.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/0cbfad04c91eb780320c340bc5a3460692d5cac4).